docs/modules/extensions/pages/processors/tree-processor.adoc

```diff
@@ -1,5 +1,38 @@
-= Tree Processor Extension Example
-:navtitle: Tree Processor
+= Tree Processor
+
+TreeProcessor is a DocumentProcessor
+
+== DSL methods
+
+include::page$processor.adoc[tag=docproc]
+
+== Process method
+
+`def process document`
+
+The process method is called from a Document after parsing.
+Therefore document attributes are available and the block structure of the document is available.
+
+The `process` method may modify the document and return the document or nil, or return a new Document.
+The document attributes are reset to the header attributes before the first tree processor is called.
+A tree processor should thus assure that the attributes of the original or returned document are reset to the header attributes by calling `document.restore_attributes` if the attributes have been modified.
+
+There are two basic strategies for a TreeProcessor.
+
+=== Query the document blocks using the `document.find_by` method
+
+This provides quick simple access to the blocks of interest, but does not provide access to the attributes set inline in the document.
```

Comment:
```suggestion
This provides quick simple access to the blocks of interest, but does not provide access to the attributes set inside in the document.
```

We can't set attributes inline can we? I think we're talking about setting attributes outside the document header, is that right?

Reply:
"set in the document after the document header"?

Reply:
> but does not provide access to the attributes set inside in the document.

This should be framed as "does not honor attributes that are set in the body of the document".